rocAL/include/readers/image/numpy_data_reader.h

```diff
@@ -0,0 +1,149 @@
+    size_t _padded_samples = 0;
+    bool _loop;
+    bool _shuffle;
+    int _read_counter = 0;
+    unsigned _seed = 0;
+    //!< _file_count_all_shards total_number of files in to figure out the max_batch_size (usually needed for distributed training).
+    size_t _file_count_all_shards;
+    std::mutex _cache_mutex;
+    std::map<std::string, NumpyHeaderData> _header_cache;
+    const RocalTensorDataType get_dtype(const std::string& format);
+    inline void skip_spaces(const char*& ptr);
+    void parse_header_data(NumpyHeaderData& target, const std::string& header);
+    template <size_t N>
+    void skip_char(const char*& ptr, const char (&what)[N]);
+    template <size_t N>
+    bool try_skip_char(const char*& ptr, const char (&what)[N]);
+    template <size_t N>
+    void skip_field(const char*& ptr, const char (&name)[N]);
+    template <typename T = int64_t>
+    T parse_int(const char*& ptr);
+    std::string parse_string(const char*& input, char delim_start = '\'', char delim_end = '\'');
+    void parse_header(NumpyHeaderData& parsed_header, std::string file_path);
+    template <typename T>
+    size_t parse_numpy_data(T* buf, std::vector<unsigned> strides, std::vector<unsigned> shapes, unsigned dim = 0);
+    bool get_header_from_cache(const std::string& file_name, NumpyHeaderData& target);
+    void update_header_cache(const std::string& file_name, const NumpyHeaderData& value);
+    void incremenet_read_ptr();
+    void increment_curr_file_idx();
+    int release();
+    void fill_last_batch();
+    void replicate_last_batch_to_pad_partial_shard();
+    std::shared_ptr<MetaDataReader> _meta_data_reader = nullptr;
+    //! Pair containing the last batch policy and pad_last_batch_repeated values for deciding what to do with last batch
+    ShardingInfo _last_batch_info = ShardingInfo();  // The members of ShardingInfo determines how the data is distributed among the shards and how the last batch is processed by the pipeline.
+    size_t _last_batch_padded_size = 0;              // The size of number of padded samples in the last batch
+    size_t _num_padded_samples = 0;                  //! Number of samples that are padded in the last batch which would differ for each shard.
+    bool _stick_to_shard = false;
+    bool _pad_last_batch_repeated = false;
+    Reader::Status generate_file_names();            // Function that would generate _file_names containing all the samples in the dataset
+    void compute_start_and_end_idx_of_all_shards();  // Start Idx of all the Shards
+    size_t get_dataset_size();                       // DataSet Size
+    size_t actual_shard_size_without_padding();      // Actual Number of Files present in the shard (without padding)
+    size_t largest_shard_size_without_padding();     // The size of the shard having largest files (without padding)
+    //!< Used to advance to the next shard's data to increase the entropy of the data seen by the pipeline>
+    void increment_shard_id();
+};
```

Comment:
Add an EOF

Reply:
Shows EOF in vscode but not in github changes